src/umode.rs

```diff
@@ -231,81 +232,107 @@ global_asm!(
 pub enum Error {
     /// Received an unexpected trap while running Umode.
     UnexpectedTrap,
-    /// Task already active.
-    TaskBusy,
+    /// Umode called panic.
+    Panic,
```

Comment:
Should we poison the U-mode task after it panics? Or if it hits an unexpected trap?

Reply:
This is where the outstanding patch of reset comes along. Reset will be called exactly when unexpected trap, panic (or even unknown hypcall) is detected as an error after `task.run()` in `send_req`.

My plan so far is the following:
1. `umode-api` and basic umode handling lifetime <- this patch
2. Expanding `umode-api` with ability to map pages in input/output when running `umode`
3. implement reset on panic/error.

After this, the feature should be completed. A baseline patch for 3 is already completed, so I could anticipate it.

Reply:
No changes regarding this. (still agree)

Reply:
Maybe drop a TODO to indicate that this is forthcoming?

Reply:
Done. Things are slightly more complex that the comment but that's the essence.